app/services/image_upload_presigned_url_generator.rb

```diff
@@ -16,6 +16,10 @@ def presigned_image_upload_url(image_type:, transaction_id:)
     end
   end
 
+  def bucket
+    super(prefix: bucket_prefix)
+  end
```

Comment:
Ruby mixins are super weird... I'm a little worried this will mess up the presigned URL generation when that calls bucket with an argument (since now the method doesn't take an argument), what if we made this a different method like `bucket_domain` or `bucket_cors_url` and use that in the initializer?

Reply:
> Ruby mixins are super weird... I'm a little worried this will mess up the presigned URL generation when that calls bucket with an argument (since now the method doesn't take an argument), what if we made this a different method like `bucket_domain` or `bucket_cors_url` and use that in the initializer?

Yeah, I share that concern. Overriding seemed like a nice fit at first, though between the arguments mismatch and the fact that we're using mixins and not extending a class, it might not be the best idea after all.

I'll tinker with it.

Reply:
`s3_resource.bucket` has a `url` method, which seems to be a good fit for this:

https://docs.aws.amazon.com/sdk-for-ruby/v2/api/Aws/S3/Bucket.html#url-instance_method

Also because it will generate URLs for CSP more closely aligned to how they're generated for the presigned URL. I'd been a bit uncertain about this previously, since buckets can be referenced both by subdomains and by path.

Updated in 328d58b. The path joining doesn't seem the cleanest, but `URI` utilities felt to me rather unpredictable in their treatment of absent trailing slash, opening up some potential unwanted behavior:

```
$ irb
irb(main):001:0> URI.join('https://s3.us-west-2.amazonaws.com/login-gov-idp-doc-capture-test.123456789-us-west-2','./*').to_s
=> "https://s3.us-west-2.amazonaws.com/*"
```

Arguably `bucket_url` might be a nice thing to have in the mixin, but then it resurfaces the original problem. Another option could be to have it _only_ in the mixin and take the bucket name as an argument:

```rb
generator = ImageUploadPresignedUrlGenerator.new
generator.bucket_url(generator.bucket(generator.bucket_prefix)))
```

🤷